java/ql/src/Telemetry/DatabaseQualityDiagnostics.ql

```diff
@@ -0,0 +1,40 @@
+/**
+ * @name Low Java analysis quality
+ * @description Low Java analysis quality
+ * @kind diagnostic
+ * @id java/diagnostic/database-quality
+ */
+
+import java
+import DatabaseQuality
+
+private newtype TDbQualityDiagnostic =
+  TTheDbQualityDiagnostic() {
+    exists(float percentageGood |
+      CallTargetStatsReport::percentageOfOk(_, percentageGood)
+      or
+      ExprTypeStatsReport::percentageOfOk(_, percentageGood)
+    |
+      percentageGood < 95
+    )
+  }
+
+class DbQualityDiagnostic extends TDbQualityDiagnostic {
+  string toString() {
+    result =
```

Comment:
Would you consider dropping a link to the troubleshooting docs where we can expand on the various failure scenarios and provide more prescriptive guidance?

Reply:
Agree about linking to our build mode docs! We do have a [Troubleshooting code scanning](https://docs.github.com/en/code-security/code-scanning/troubleshooting-code-scanning) section of the GitHub Docs where we *could* add an article later on that discusses what this warning means, and what could be wrong, if you think it would be needed. But that could come later on, we can create an issue in [docs-content](https://github.com/github/docs-content) repo for it.